File: servers/assessment/assessmentSchemas/assessmentSchemaDTO/assessmentSchema.go
package assessmentSchemaDTO

import (
	"time"

	"github.com/google/uuid"
	db "github.com/prompt-edu/prompt/servers/assessment/db/sqlc"
)

type AssessmentSchema struct {
	ID          uuid.UUID `json:"id"`
	Name        string    `json:"name"`
	Description string    `json:"description"`
	CreatedAt   time.Time `json:"createdAt"`
	UpdatedAt   time.Time `json:"updatedAt"`
	// IsOwnedByCurrentPhase is only populated by phase-scoped listings; nil elsewhere.
	IsOwnedByCurrentPhase *bool `json:"isOwnedByCurrentPhase,omitempty"`
}

func MapDBAssessmentSchemaToDTOAssessmentSchema(dbSchema db.AssessmentSchema) AssessmentSchema {
	return AssessmentSchema{
		ID:          dbSchema.ID,
		Name:        dbSchema.Name,
		Description: dbSchema.Description.String,
		CreatedAt:   dbSchema.CreatedAt.Time,
		UpdatedAt:   dbSchema.UpdatedAt.Time,
	}
}

File: clients/assessment_component/src/assessment/interfaces/assessmentSchema.ts
export interface AssessmentSchema {
  id: string
  name: string
  description: string
  createdAt?: Date
  updatedAt?: Date
  isOwnedByCurrentPhase?: boolean
}

export interface CreateAssessmentSchemaRequest {
  name: string
  description: string
}

export interface UpdateAssessmentSchemaRequest {
  name: string
  description: string
}

export interface CreateOrUpdateAssessmentSchemaCoursePhaseRequest {
  assessmentSchemaID: string
  coursePhaseID: string
}
